### PR TITLE
Synthesizer skips over embeddings file if model only has one speaker

### PR DESCRIPTION
**Pull request by wonkothesanest**
It looks like the Neon model is special in that t does not have a speaker_name and it wants to get the only item available. This was blocking a valid model with one speaker and a d_vector_file from being executed to get the embedding.

It looks like the Neon model is special in that t does not have a speaker_name and it wants to get the only item available.
This was blocking a valid model with one speaker and a d_vector_file from being executed to get the embedding.

Leaving the check to be second still works fine for the case that there is only one model and no speaker name.

This is in reference to bug 2320
https://github.com/coqui-ai/TTS/issues/2320

(updated from  trying to get my author right)

### CODE REVIEW

**Comment by wonkothesanest** [comment]
Can someone please guide me to some documentation on how to get the tests to pass? I believe there is not a correct token setup for Coqui TTS.
"No API token found for 🐸Coqui Studio voices - https://coqui.ai/ \n"

File: TTS/utils/synthesizer.py [hunk 1]
```diff
@@ -266,7 +262,9 @@ def tts(
                 else:
                     # get speaker idx from the speaker name
                     speaker_id = self.tts_model.speaker_manager.name_to_id[speaker_name]
-
+            # handle Neon models with single speaker.
+            elif len(sef.tts_model.speaker_manager.name_to_id) == 1:
```
Comment (erogol) [inline TTS/utils/synthesizer.py:266]: sef -> self.

**Review by erogol — CHANGES_REQUESTED** [review]

File: TTS/utils/synthesizer.py (hunk 1 above)
Comment (wonkothesanest) [reply TTS/utils/synthesizer.py]: wow that's embarrassing 😬

**Comment by wonkothesanest** [comment]
I have 5 failing tests all of which seem to have to do with not having a Coqui Studio Token.  Can someone please help me understand how I might be able to fix this issue?  I don't see where I could set the environment to add this token.

Relevant error message:
```
> Generator Model: multiband_melgan_generator
 > Discriminator Model: melgan_multiscale_discriminator
 > Text: This is an example.
 > Text splitted to sentences.
['This is an example.']
 > Processing time: 0.34621286392211914
 > Real-time factor: 0.15287555369838846
 > Saving output to /home/runner/work/TTS/TTS/tests/outputs/output.wav
 > Using 🐸Coqui Studio model:  coqui_studio/en/Torcull Diarmuid/coqui_studio
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.16/x64/bin/tts", line 8, in <module>
    sys.exit(main())
  File "/opt/hostedtoolcache/Python/3.7.16/x64/lib/python3.7/site-packages/TTS/bin/synthesize.py", line 330, in main
    api = TTS(model_name=args.model_name)
  File "/opt/hostedtoolcache/Python/3.7.16/x64/lib/python3.7/site-packages/TTS/api.py", line 285, in __init__
    self.load_tts_model_by_name(model_name, gpu)
  File "/opt/hostedtoolcache/Python/3.7.16/x64/lib/python3.7/site-packages/TTS/api.py", line 375, in load_tts_model_by_name
    self.csapi = CS_API()
  File "/opt/hostedtoolcache/Python/3.7.16/x64/lib/python3.7/site-packages/TTS/api.py", line 67, in __init__
    self._check_token()
  File "/opt/hostedtoolcache/Python/3.7.16/x64/lib/python3.7/site-packages/TTS/api.py", line 94, in _check_token
    "No API token found for 🐸Coqui Studio voices - https://coqui.ai/ \n"
ValueError: No API token found for 🐸Coqui Studio voices - https://coqui.ai/ 
Visit 🔗https://app.coqui.ai/account to get one.
Set it as an environment variable `export COQUI_STUDIO_TOKEN=<token>`

FAIL
```

**Comment by wonkothesanest** [comment]
@erogol Anything else I need to do here for this PR?

**Comment by erogol** [comment]
@wonkothesanest thx for the PR. I'll review it as soon as I can

**Comment by wonkothesanest** [comment]
@erogol 
Anything else needed from me on this pr?

**Pull request merged**